jmclient/jmclient/taker.py

```diff
@@ -557,6 +510,84 @@ def receive_utxos(self, ioauth_data):
         return (True, list(self.maker_utxo_data.keys()),
                 bintohex(self.latest_tx.serialize()))
 
+    def _verify_ioauth_data(self, ioauth_data):
+        verified_data = []
+        # Need to authorize against the btc pubkey first.
+        for nick, nickdata in ioauth_data.items():
+            utxo_list, auth_pub, cj_addr, change_addr, btc_sig, maker_pk = nickdata
+            if not self.auth_counterparty(btc_sig, auth_pub, maker_pk):
+                jlog.debug(
+                    "Counterparty encryption verification failed, aborting: " + nick)
+                # This counterparty must be rejected
+                continue
+
+            if not validate_address(cj_addr)[0]\
+                    or not validate_address(change_addr)[0]:
+                jlog.warn("Counterparty provided invalid address: {}".format(
+                    (cj_addr, change_addr)))
+                # Interpreted as malicious
+                self.add_ignored_makers([nick])
+                continue
+
+            try:
+                maker_inputs_data = self._verify_ioauth_inputs(
+                    nick, utxo_list, auth_pub)
+            except IoauthInputVerificationError as e:
+                for msg in e.messages:
+                    jlog.warning(msg)
```

Comment:
Doesn't this duplicate the warning messages?
We already log them in `_verify_ioauth_inputs`, no?

```python
        if change_amount < jm_single().DUST_THRESHOLD:
            fmt = ('ERROR counterparty requires sub-dust change. nick={}'
                   ' totalin={:d} cjamount={:d} change={:d}').format
            jlog.warn(fmt(nick, total_input, self.cjamount, change_amount))
            jlog.warn("Invalid change, too small, nick= " + nick)
```

Reply:
Thanks for testing and reviewing! I've missed to delete those logging lines. Done in 0c3cc6cb227f625e56748428f2b2a20aa385866f